tests/mli/test_error_handling.py

```diff
@@ -0,0 +1,164 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import multiprocessing as mp
+
+import pytest
+
+from smartsim._core.mli.infrastructure.control.workermanager import (
+    WorkerManager,
+    exception_handler,
+)
+from smartsim._core.mli.infrastructure.worker.worker import InferenceReply
+from smartsim._core.mli.message_handler import MessageHandler
+
+from .channel import FileSystemCommChannel
+from .featurestore import FileSystemFeatureStore
+from .worker import IntegratedTorchWorker
+
+# The tests in this file belong to the group_a group
+pytestmark = pytest.mark.group_a
+
+
+@pytest.fixture
+def setup_worker_manager(test_dir):
+    work_queue: "mp.Queue[bytes]" = mp.Queue()
+    integrated_worker = IntegratedTorchWorker()
+    file_system_store = FileSystemFeatureStore(test_dir)
+
+    worker_manager = WorkerManager(
+        work_queue,
+        integrated_worker,
+        file_system_store,
+        as_service=False,
+        cooldown=10,
```

Comment:
- `cooldown=10` tells the service to run 10s after it satisfies shutdown conditions

this _might_ cause tests to run for an extra 10s, even if it's not necessary. If these tests are taking a long time, consider adjusting cooldown